### DISCUSSION TITLE
code server not installing on raspberry pi

### DISCUSSION THREAD

**Discussion by rlarudqls630** [coder/code-server#4379] [Q&A]
I am trying to install code server on raspberry pi but after running 
sudo curl -fsSL https://code-server.dev/install.sh | sh
this line, I am getting some weired message.

Done in 448.73s.

npm package has been installed.

Extend your path to use code-server:
  PATH="$(yarn global bin):$PATH"
Then run with:
  code-server

It says success, but I am not able to run the code server using code-server command. 

please hlep
thank you.

**Comment by Bickerchick739** [comment]
Can you please stop it cuz I know everything that u been doing on my
account either stop it or things r not going to look good for you in the
long run u done enough to me so stay off my shit an I like it cuzur
bisexual an thanks for all the info that you been doing I'm so glad now I
look at this phone because u have my iphone connect to this phone an didn't
know body do it but u

**Comment by lg-nineoneninethreenine** [comment]
@jsjoeio can you remove the really abusive message? Doesn't look good here. And as an answer, there is a bug with one of the dependencies so you can't install code-server yet... Tracked in another discussion, so there is a kinda big and I am waiting too, so Raspberry Pi users gotta wait.

**Reply by rlarudqls630** [reply]
thx 
i thought this github was clean community....
thx for traking down the answer as well 
hope this issue gets resolved soon.

**Reply by jsjoeio** [reply]
@im-coder-lg do we think this is that same argon2 error?

I wonder if we could mark it as an optional dependency as a workaround? I would love to unblock Raspberry Pi users

**Reply by lg-nineoneninethreenine** [reply]
Hmm... Maybe @rlarudqls630 did the same as @Slodziak190q, `npm i code-server` or `yarn add code-server`  instead of `npm i -g code-server` or yarn add global code-server`(maybe even `yarn global add code-server`). But still it's blocked.
>I wonder if we could mark it as an optional dependency as a workaround?

Is it possible? I think it might screw up something. If it's a problem with Argon2, we gotta tell them and get a simple solution. By the way, what is Argon2 used for here?
> I would love to unblock Raspberry Pi users

Same here. I can't find a valid reason to use my Raspberry Pi, other than checking simple DEs, like MATE, Xfce. If this happens, I could try the new Link features, and even make some websites from that.

**Reply by lg-nineoneninethreenine** [reply]
And, yeah, I have a small idea. Can we make a discussion about public issues known to the maintainers so that people can know that there is already a Public Issues discussion with their(kinda sometimes new bugs can occur) bug. We can make a to-do list and loop the bug there, so it will display the bug name too, and if solved, we can cross that out(the `- [x]` in markdown) and we can inform people that the bug is solved.

**Reply by Slodziak190q** [reply]
> Hmm... Maybe @rlarudqls630 did the same as @Slodziak190q, `npm i code-server` or `yarn add code-server` instead of `npm i -g code-server` or yarn add global code-server`(maybe even `yarn global add code-server`). But still it's blocked.
> 
> > I wonder if we could mark it as an optional dependency as a workaround?
> 
> Is it possible? I think it might screw up something. If it's a problem with Argon2, we gotta tell them and get a simple solution. By the way, what is Argon2 used for here?
> 
> > I would love to unblock Raspberry Pi users
> 
> Same here. I can't find a valid reason to use my Raspberry Pi, other than checking simple DEs, like MATE, Xfce. If this happens, I could try the new Link features, and even make some websites from that.



Problem with all raspberry pi is that rpi can only run code-server 3.10.0 so @rlarudqls630 should run the command 
sudo npm install -g code-server@3.10.0 --unsafe-perm=true --allow-root
it worked for me

**Reply by rlarudqls630** [reply]
I am currently running it different way. 
Windows are allowing users to download Ubuntu on the system. 
So I installed Ubuntu and running cod-server in the background.

I guess you can do the same thing with raspberry pi as well. 
I believe that you can install Ubuntu on raspberry pi.

**Reply by lg-nineoneninethreenine** [reply]
> I am currently running it different way. 
> Windows are allowing users to download Ubuntu on the system. 
> So I installed Ubuntu and running cod-server in the background.
> 
> I guess you can do the same thing with raspberry pi as well. 
> I believe that you can install Ubuntu on raspberry pi.

Dude, it ain't gonna work. Since on Raspberry Pis, you need to use NPM only. The deb installation file wornt work since Raspberry Pis are 32-bit systems.

**Comment by jsjoeio** [comment]
> Is it possible? I think it might screw up something. If it's a problem with Argon2, we gotta tell them and get a simple solution. 

Maybe! Well I did a quick search and found [this](https://github.com/ranisalt/node-argon2/issues/43). Maybe you have to build from source on Pi to get it working. If someone wants to try, that would be awesome. 

> By the way, what is Argon2 used for here?

Hashing passwords! Well if I recall correctly, we store the password hashed instead of plaintext. 

> Can we make a discussion about public issues known to the maintainers so that people can know that there is already a Public Issues discussion with their(kinda sometimes new bugs can occur) bug. We can make a to-do list and loop the bug there, so it will display the bug name too, and if solved, we can cross that out(the - [x] in markdown) and we can inform people that the bug is solved.

Fantastic idea! Do you mind writing an example here and then I can create a new Discussion, tag you and pin it?

**Reply by lg-nineoneninethreenine** [reply]
Well, I can't since I am busy(that's why the mention would have showed the tag) but I will try it by the time I can use my laptop(I don't have it right now) but if you can, list out a common bug and add it in the checklist like this:
```markdown
-[ ] #issuenumber
```
That has to do the job. The rest can be curated by you. I will check every few hours so I can't respond quickly as I used to.

**Reply by lg-nineoneninethreenine** [reply]
Well, can you try on a safe Gitpod environment(building Argon2 from source)? If it works there, it might work here too! I promised an Asciicast but can't do that till I get my laptop.

If you are up for that, visit the Argon2 repo, and in the URL, don't change anything, just add `gitpod.io/#/` and the URL. Then log in via GitHub, give the required perms and then you are in charge.

**Reply by jsjoeio** [reply]
> If you are up for that, visit the Argon2 repo, and in the URL, don't change anything, just add gitpod.io/#/ and the URL. Then log in via GitHub, give the required perms and then you are in charge.

Oh good to know! Okay this Discussion is getting long so I think I'll summarize investigating argon2 in a new issue and we'll track there.

**Reply by jsjoeio** [reply]
Tracking here: https://github.com/cdr/code-server/discussions/4379#discussioncomment-1571571

**Reply by jsjoeio** [reply]
Started a discussion here: https://github.com/cdr/code-server/discussions/4423
